chain/chain/src/runtime/tests.rs

```diff
@@ -1686,14 +1685,11 @@ fn test_prepare_transactions_storage_proof() {
     .unwrap();
 
     assert_eq!(proposed_transactions.transactions.len(), transactions_count);
-    assert!(proposed_transactions.storage_proof.is_some());
 
     let validator_storage_config = RuntimeStorageConfig {
         state_root: env.state_roots[0],
         use_flat_storage: true,
-        source: StorageDataSource::Recorded(PartialStorage {
-            nodes: proposed_transactions.storage_proof.unwrap(),
-        }),
+        source: StorageDataSource::Recorded(PartialStorage::default()),
```

Comment:
This is obviously not the correct change.  I just added it here to make things compile.  This test is currently failing.  I am not sure how to fix the issue though.  I need to somehow need to do `trie.recorded_storage().map(|s| s.nodes)` from the call to `prepare_transactions` above but I am not sure how exactly to accomplish that.  Any help is much appreciated.

Reply:
Why was `storage_proof` field removed from the `PreparedTransactions` in the first place?

Reply:
@nagisa and @wacban: based on my analysis, the field was only used to set the `transactions_storage_proof` argument to `create_state_witness` [here](https://github.com/near/nearcore/pull/13185/files#diff-5fe9869228ee4876b7d1cdcdcf40ffd3e4582c03cd70bb0345a7ce81246084feL124).  So AFAICT, there is no consumer of this field in production code.

Reply:
Ah, was the proof only used to validate new transactions? Just for completeness, why is it that the old transaction not require the same? Is that because the old transaction validation happens in the main transition and is captured in the storage proof there? This would make sense assuming the transaction validation happens from within apply where it would be recorded and automatically verified by chunk validators.

Reply:
In that case however I don't see why the empty storage breaks the test though.

Reply:
for posterity, we decided that the test is now obsolete and removed it.